BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BTConnector.java

```diff
@@ -46,45 +43,38 @@ public WifiBtStatus(){
     }
 
     public interface  Callback{
-        public void Connected(BluetoothSocket socket, boolean incoming,String peerId,String peerName,String peerAddress);
-        public void ConnectionFailed(String peerId,String peerName,String peerAddress);
-        public void StateChanged(State newState);
+        void Connected(BluetoothSocket socket, boolean incoming,String peerId,String peerName,String peerAddress);
+        void ConnectionFailed(String peerId,String peerName,String peerAddress);
+        void StateChanged(State newState);
     }
 
     public interface  ConnectSelector{
-        public ServiceItem CurrentPeersList(List<ServiceItem> available);
-        public void PeerDiscovered(ServiceItem service);
+        ServiceItem CurrentPeersList(List<ServiceItem> available);
+        void PeerDiscovered(ServiceItem service);
     }
 
-    private State myState = State.NotInitialized;
-
-    WifiBase mWifiBase = null;
-    BTConnector_Discovery mBTConnector_Discovery = null;
-
-    BluetoothBase mBluetoothBase = null;
-    BTConnector_BtConnection mBTConnector_BtConnection = null;
+    private WifiBase mWifiBase = null;
+    private BTConnector_Discovery mBTConnector_Discovery = null;
 
+    private BluetoothBase mBluetoothBase = null;
+    private BTConnector_BtConnection mBTConnector_BtConnection = null;
 
     private String mInstanceString = "";
-    static String JSON_ID_PEERID   = "pi";
-    static String JSON_ID_PEERNAME = "pn";
-    static String JSON_ID_BTADRRES = "ra";
+    static final String JSON_ID_PEERID   = "pi";
+    static final String JSON_ID_PEERNAME = "pn";
+    static final String JSON_ID_BTADRRES = "ra";
 
-
-    boolean isStarting = false;
     private Callback callback = null;
     private ConnectSelector connectSelector = null;
     private Context context = null;
     private Handler mHandler = null;
 
-
-    BTConnectorSettings ConSettings = new BTConnectorSettings();
+    private BTConnectorSettings ConSettings = new BTConnectorSettings();
 
     public BTConnector(Context Context, Callback Callback, ConnectSelector selector, BTConnectorSettings settings){
         this.context = Context;
         this.callback = Callback;
         this.mHandler = new Handler(this.context.getMainLooper());
-        this.myState = State.NotInitialized;
         this.ConSettings = settings;
         this.connectSelector = selector;
     }
```

Comment:
Both the start and stop methods seem like the kind of methods that should be called exactly once apiece. Once stop is called shouldn't the BTConnector object be garbage collected and a new one created? This would make the logic much simpler.